src/operator/bilinear_sampler.cc

```diff
@@ -28,7 +28,7 @@
 namespace mshadow {
 template<typename DType>
 bool between(DType value, int lowerBound, int upperBound) {
-  return (value >= lowerBound && value <= upperBound);
+  return (static_cast<int>(value) >= lowerBound && static_cast<int>(value) <= upperBound);
```

Comment:
5.5 is not a valid value though. The value has to be an integer (in this file) or an unsigned integer in the other file. In such cases, there shouldn't be such an issue right?